typing_inspect.py

```diff
@@ -404,17 +499,30 @@ class Node(Generic[T]):
     return gen_type if gen_type is not None else type(obj)
 
 
-def get_generic_bases(tp):
-    """Get generic base types of a type or empty tuple if not possible.
-    Example::
+if not LEGACY_TYPING:
+    def get_generic_bases(tp):
+        """Get generic base types of a type or empty tuple if not possible.
+        Example::
 
-        class MyClass(List[int], Mapping[str, List[int]]):
-            ...
-        MyClass.__bases__ == (List, Mapping)
-        get_generic_bases(MyClass) == (List[int], Mapping[str, List[int]])
-    """
+            class MyClass(List[int], Mapping[str, List[int]]):
+                ...
+            MyClass.__bases__ == (List, Mapping)
+            get_generic_bases(MyClass) == (List[int], Mapping[str, List[int]])
+        """
+
+        return getattr(tp, '__orig_bases__', ())
+else:
+    def get_generic_bases(tp):
+        """Get generic base types of a type or empty tuple if not possible.
+        Example::
 
-    return getattr(tp, '__orig_bases__', ())
+            class MyClass(List[int], Mapping[str, List[int]]):
+                ...
+            MyClass.__bases__ == (List, Mapping)
+            get_generic_bases(MyClass) == (List[int], Mapping[str, List[int]])
+        """
+
+        return tuple(t for t in tp.__bases__ if isinstance(t, GenericMeta))
```

Comment:
Again, put just the body inside the if/else (it looks like you try to optimize performance, but I don't think this is actually important).

Reply:
ok